rover/__test__/handler/block.test.ts

```diff
@@ -0,0 +1,27 @@
+import { blockHandler } from "../../src/handler/block";
+
+let blockNumber = 42;
+let blockDetails = { transactions: [1, 2, 3] };
+let provider = {
+  getBlock: jest.fn().mockResolvedValue(blockDetails)
+};
+
+let storage = { storeBlock: jest.fn() };
+let emitter = { emit: jest.fn() };
+
+describe("blockHandler", () => {
+  it("stores the block information", async () => {
+    await blockHandler(storage, blockNumber, emitter, provider);
+    expect(storage.storeBlock).toHaveBeenCalled();
```

Comment:
could you use `toHaveBeenCalledWith` here?

Reply:
At the moment this is being avoided because it is being called with the result of a mock that I built